core/src/main/scala/org/broadinstitute/dsde/rawls/billing/GoogleBillingProjectCreator.scala

```diff
@@ -0,0 +1,42 @@
+package org.broadinstitute.dsde.rawls.billing
+
+import akka.http.scaladsl.model.StatusCodes
+import org.broadinstitute.dsde.rawls.dataaccess.{GoogleServicesDAO, SamDAO}
+import org.broadinstitute.dsde.rawls.model.{CreateRawlsV2BillingProjectFullRequest, ErrorReport, ErrorReportSource, SamResourceTypeNames, SamServicePerimeterActions, ServicePerimeterName, UserInfo}
+import org.broadinstitute.dsde.rawls.user.UserService.syncBillingProjectOwnerPolicyToGoogleAndGetEmail
+
+import java.net.URLEncoder
+import scala.concurrent.{ExecutionContext, Future}
+import java.nio.charset.StandardCharsets.UTF_8
+
+class GoogleBillingProjectCreator(samDAO: SamDAO, gcsDAO: GoogleServicesDAO)(implicit executionContext: ExecutionContext) extends BillingProjectCreator {
+  implicit val errorReportSource: ErrorReportSource = ErrorReportSource("rawls")
+
+  override def validateBillingProjectCreationRequest(createProjectRequest: CreateRawlsV2BillingProjectFullRequest, userInfo: UserInfo): Future[Unit] = {
+    for {
```

Comment:
Scala question… why is this written as a for/yield instead of simply 3 sequential steps?

Reply:
So this is again because of Futures (`checkServicePerimeterAccess` returns a `Future[Unit]`)?

Reply:
That's correct.